Tests/SonarScanner.MSBuild.PreProcessor.Test/SonarWebServiceTest.cs

```diff
@@ -1169,7 +1182,7 @@ public async Task GetProperties_RequestUrl(string hostUrl, string versionUrl, st
         private static Stream CreateCacheStream(IMessage message)
         {
             var stream = new MemoryStream();
-            message.WriteTo(stream);
+            message.WriteDelimitedTo(stream);
```

Comment:
This is an important change that's well hidden.

The stream we receive from the server contains a sequence of objects (not a single one), and this sequence does not have a parent. In order to correctly emulate the serialization/deserialization and make it work properly, we have to use `WriteDelimitedTo` instead of `WriteTo`.